src/env.c

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdlib.h>
+
+#include "rcutils/env.h"
+#include "rcutils/error_handling.h"
+
+// TODO(cottsay): Move the stuff in get_env.c in here
+
+bool
+rcutils_set_env(const char * env_name, const char * env_value)
+{
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
+    env_name, "env_name is null", return false);
+
+#ifdef _WIN32
+  if (NULL == env_value) {
+    env_value = "";
+  }
+  if (0 != _putenv_s(env_name, env_value)) {
+    return false;
+  }
+#else
+  if (NULL == env_value) {
+    if (0 != unsetenv(env_name)) {
+      return false;
```

Comment:
Instead of just swallowing the error here, I think it would be a better idea to call strerror to get the actual error and then `RCUTILS_SET_ERROR_MSG...` to set it.  The same goes above (though the call on Windows is different) and below.

Reply:
Do you really think it's worth creating a buffer to receive the error message? The docs for all of these functions say that the only reasons they fail are if an invalid key was specified or there was a memory allocation error.

I added a message with the error code. If you'd really rather see the strerror, I can do that instead.